lib/rpmsg/rpmsg_virtio.c

```diff
@@ -384,11 +384,9 @@ static void *rpmsg_virtio_get_tx_payload_buffer(struct rpmsg_device *rdev,
 		 * use metal_sleep_usec() method by default.
 		 */
 		status = rpmsg_virtio_notify_wait(rvdev, rvdev->rvq);
-		if (status == RPMSG_EOPNOTSUPP) {
+		if (status != RPMSG_SUCCESS) {
```

Comment:
> This looks like a valid bug fix. The only change I would request is to remove the line in the commit message about redundant continue logic, since it doesn't seem to be relevant any more.

You are right, the second line commit message has been removed.